catalogue/webapp/services/wellcome/index.ts

```diff
@@ -55,43 +33,39 @@ export const catalogueApiError = (): CatalogueApiError => ({
 //
 // As such, we use an agent which is configured to expire free sockets after 59s
 // A good explanation of the problem, as well as the solution, is available here:
+
 // https://connectreport.com/blog/tuning-http-keep-alive-in-node-js/
 const agentKeepAlive = new Agent({
   keepAlive: true,
   freeSocketTimeout: 1000 * 59, // 1s less than the akka-http idle timeout
 });
 
-export const catalogueFetch = (
+export const wellcomeApiFetch = (
   url: string,
   options?: Record<string, string>
 ): Promise<Response> => {
   return fetch(url, { ...options, agent: agentKeepAlive });
 };
 
-export type QueryProps<Params> = {
-  params: Params;
-  pageSize?: number;
-  toggles: Toggles;
-};
-
-export async function catalogueQuery<Params, Result extends ResultType>(
-  endpoint: string,
-  { params, toggles, pageSize }: QueryProps<Params>
-): Promise<CatalogueResultsList<Result> | CatalogueApiError> {
-  const apiOptions = globalApiOptions(toggles);
-  const extendedParams = {
-    ...params,
-    pageSize,
-  };
-
-  const searchParams = new URLSearchParams(
-    propsToQuery(extendedParams)
-  ).toString();
+export const wellcomeApiError = (): WellcomeApiError => ({
+  errorType: 'http',
+  httpStatus: 500,
+  label: 'Internal Server Error',
+  description: '',
+  type: 'Error',
+});
 
-  const url = `${rootUris[apiOptions.env]}/v2/${endpoint}?${searchParams}`;
+export type WellcomeApiError = {
+  errorType: string;
+  httpStatus: number;
+  label: string;
+  description: string;
+  type: 'Error';
+};
 
+export const wellcomeApiQuery = async (url: string) => {
   try {
-    const res = await catalogueFetch(url);
+    const res = await wellcomeApiFetch(url);
```

Comment:
Yes!!! This is exactly the sort of thing I meant, push all the error handling into a single function.